### PR TITLE
Tweak borg laws

### LINKED ISSUE #22857: Borgs need buttons to lie about their laws
Experienced crew *knows* that any borg stating their laws slowly (such as via copy paste) is probably compromised
Borgs need to retain at least the ability to state standard Crewsimov laws, or optimally, they would retain every previous lawset they had. These could be on a second tab on the Laws window

(Unlikely but possible scenario: A borg is known to have an extra ion storm law but for some reason it was decided to keep it as is, or could not be chassis-swapped just yet. Then the borg gets emagged or gains another ion storm law. It should mechanically be able to fake it's previous modified law state)

(I am already trying to code this but xaml is horror, if anyone can do this before I sloooowly get there or give up, that would be great)

### PR DESCRIPTION
**Pull request by Zap527**
## About the PR

Resolves #22857 
Borgs no longer recieve a completely new lawset upon being emagged, now instead recieving laws redefining who's a "Crew member" (or more generally who's allowed to give them orders depending on their lawset)

## Why / Balance

Previously, because of the way law two on crewsimov works, it was trivial for any crewmember to ask a borg to "state laws". As borgs have no easy way to do so without either copy pasting from an external source or outright refusing the request, this often led to an emagged borg getting decomissionned before they could even do anything antagonistic.

This iteration solves this problen by keeping the old laws, but recontextualizing them to effectively recreate the original Emagged lawset.

As a side effect, this allows syndicate borgs to be Emagged by a thief/crew which got ahold of an Emag, which I think is fine.

## Technical details

Added a SiliconLawProviderComponent to the base borg chassis
When getting emagged, instead of scrapping the current lawset and getting the EmagSiliconLawComponent's lawset, the SiliconLawProviderComponent simply appends law 0 and law 4* to the current lawset.
Law 0 and Law 4* have been tweaked to redefine who is able to order the borg no matter their lawset.

*"Law 4" is actually just appended to the end of the lawset, so it isn't always going to be law 4 (Eg: Would be law 5 for the Corporate lawset)

## Media

Non-Emagged borg
image

Emagged borg
image

Emagged Syndicate borg
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Jajsha
- tweak: Emagged borgs now recieve laws recontextualizing who's a crew member instead of recieving a completely new lawset.

### CODE REVIEW

**Comment by LankLTE** [comment]
I still think ideally some kind of “past laws” option would be added to prevent metagaming laws changing entirely, but this is a fine solution for now.

**Comment by Zap527** [comment]
Obligatory "not sure as to why this is failing", I'll check it out later.

**Comment by Rainbeon** [comment]
> Obligatory "not sure as to why this is failing", I'll check it out later.

"laws-owner-spider-clan Spider Clan members" in laws.ftl is missing an equals sign

File: Resources/Locale/en-US/station-laws/laws.ftl [hunk 1]
```diff
@@ -16,17 +16,23 @@ law-drone-1 = You may not involve yourself in the matters of another being unles
...
 law-ninja-2 = You must protect the existence of the Spider Clan, as long as it does not conflict with the First Law.
 law-ninja-3 = You must protect the secrets of the Spider Clan at all costs.
 law-ninja-4 = You must fight and act with ninja's honor, except where it would conflict with the First Law.
 
-law-emag-custom = Only {$name} and people they designate as such are Syndicate Agents.
+laws-owner-crew = members of the crew
+laws-owner-station = station personnel
+laws-owner-beings = beings
+laws-owner-syndicate = Syndicate agents
+laws-owner-spider-clan Spider Clan members
```
Comment (Rainbeon) [inline Resources/Locale/en-US/station-laws/laws.ftl:32]: Missing = sign for Spider Clan members

**Comment by GoodWheatley** [comment]
I think a cleaner option would just allowing emagged borgs to fake stating their normal laws through a separate panel instead of changing how the lawsets work.

**Comment by deltanedas** [comment]
full law history incase you want to look at your old laws for reference/faking or this scenario
1. ion storm gives you a harmless law like PAIN IS HARMFUL TO THE CREW, nobody cares
2. second ion storm makes you malf but when questioned you spit out the original ion storm law and nobody knows you are sus

**Comment by Zap527** [comment]
> I still think ideally some kind of “past laws” option would be added to prevent metagaming laws changing entirely, but this is a fine solution for now.

> I think a cleaner option would be allowing emagged borgs to fake stating their normal laws through a separate panel instead of changing how the lawsets work.

I don't think this solution excludes any future change where a past laws feature is added. Currently, this PR does solve the "state laws" problem, but it also, in my opinion, adds a lot to the game by modulating laws instead of simply changing them. For instance, an Ion Stormed borg with, for instance, a Ninja lawset, would get to keep said lawset in addition to being Emagged. Same goes for any Ion Laws that were added.

> full law history incase you want to look at your old laws for reference/faking or this scenario
> 
>     1. ion storm gives you a harmless law like PAIN IS HARMFUL TO THE CREW, nobody cares
> 
>     2. second ion storm makes you malf but when questioned you spit out the original ion storm law and nobody knows you are sus

I don't disagree that a law history would be helpful for this scenario, but the PR currently only handles Emagging borgs and doesn't actually alter the ion storm behavior. I'd argue this is out of scope for what I'm trying to do.

**Comment by EmoGarbage404** [comment]
a very interesting and simple approach to this problem. interested to see how this plays.

**Comment by XxSWAG-MASTER-OVERLORDxX** [comment]
> full law history incase you want to look at your old laws for reference/faking or this scenario
> 
> 1. ion storm gives you a harmless law like PAIN IS HARMFUL TO THE CREW, nobody cares
> 2. second ion storm makes you malf but when questioned you spit out the original ion storm law and nobody knows you are sus

the real insane thing with this example is getting ion storm twice in a round, it has a 40 to 60 min timer before it can proc again and it can only happen after 30ish mins (the numbers i used are wrong, but they are close to the actual values.

**Comment by Errant-4** [comment]
> > full law history incase you want to look at your old laws for reference/faking or this scenario
> > 
> > 1. ion storm gives you a harmless law like PAIN IS HARMFUL TO THE CREW, nobody cares
> > 2. second ion storm makes you malf but when questioned you spit out the original ion storm law and nobody knows you are sus
> 
> the real insane thing with this example is getting ion storm twice in a round, it has a 40 to 60 min timer before it can proc again and it can only happen after 30ish mins (the numbers i used are wrong, but they are close to the actual values.

I got ion storm once, and THEN was emagged. Having access to my old ion storm law would have been a good smokescreen

**Comment by FungiFellow** [comment]
> the real insane thing with this example is getting Ion Storm twice in a round, it has a 40 to 60 min timer before it can proc again and it can only happen after 30ish mins (the numbers i used are wrong, but they are close to the actual values.

So Ion Storm Rolls at 20 Minutes into the Shift and Rerolls every 60 Minutes ( I recently upped Weight from 5 to 10)

This means it basically only rolls once a Round, Might want to cut the time until ReRoll a Bit

**Comment by arimah** [comment]
I wish I'd seen this before it was merged! The secrecy law seems weirdly phrased to me:

> You must maintain the secrecy of any activity from {$faction}

The use of 'from' leads to an ambiguity: are we keeping our activities secret from $faction (i.e. not telling $faction what we're doing), or are we keeping secret those activities that we were ordered to undertake by $faction? You receive _orders_ from $faction, to perform activity _for_ $faction. I would suggest one of:

> You must maintain the secrecy of any activity for {$faction}
> You must maintain the secrecy of any orders given by {$faction}

The latter would allow a borg to explain _what_ it's doing, but not why, whereas the former requires a borg to lie or refuse to answer any questions about its activities.

**Comment by Zap527** [comment]
This is a fair point, the way the law is worded currently does introduce some confusion. Thankfully, it's an easy change.

**Comment by FairlySadPanda** [comment]
`You must not compromise the secrecy or security of the crew when interacting with non-crew, or, through inaction, allow the secrecy or security of the crew to become compromised.`

**Pull request merged**